ddtrace/contrib/vertica/patch.py

```diff
@@ -0,0 +1,246 @@
+import importlib
+import wrapt
+
+import ddtrace
+from ddtrace import config, Pin
+from ddtrace.ext import net, AppTypes
+from ddtrace.utils.wrappers import unwrap
+
+from .constants import APP
+from ...ext import db as dbx
+
+"""
+Note: we DO NOT import the library to be patched at all!
+
+What this means is that this approach completely solves our patching problem as
+well. wrapt will hook into the module import system for the patches that we
+provide.
+"""
+
+_PATCHED = False
+
+def copy_before(instance, span, conf, *args, **kwargs):
+    span.set_tag("query", args[0])
+
+def execute_before(instance, span, conf, *args, **kwargs):
+    span.set_tag("query", args[0])
+
+def execute_after(result, instance, span, conf, *args, **kwargs):
+    span.set_metric(dbx.ROWCOUNT, instance.rowcount)
+
+def fetch_after(result, instance, span, conf, *args, **kwargs):
+    span.set_metric(dbx.ROWCOUNT, instance.rowcount)
+
+def cursor_after(cursor, instance, span, conf, *args, **kwargs):
+    tags = {}
+    tags[net.TARGET_HOST] = instance.options["host"]
+    tags[net.TARGET_PORT] = instance.options["port"]
+    if "user" in instance.options:
+        tags[dbx.USER] = instance.options["user"]
+    if "database" in instance.options:
+        tags[dbx.NAME] = instance.options["database"]
+
+    pin = Pin(
+        service=config.vertica["service_name"],
+        app=APP,
+        app_type=AppTypes.db,
+        tags=tags,
+        _config=config.vertica["patch"]["vertica_python.vertica.cursor.Cursor"]
+    )
+    pin.onto(cursor)
+
+
+# tracing configuration
+config._add(
+    "vertica",
+    {
+        "service_name": "vertica",
+        "app": "vertica",
+        "app_type": "db",
+        "patch": {
+            "vertica_python.vertica.connection.Connection": {
+                "routines": {
```

Comment:
We already discussed this in PM, an alternative approach could be to have:
```
"vertica_python.vertica.connection.Connection": {
  "cursor": { ... }
  "__other_metadata": ...
}
```
The pattern that I mean is not adding the `routines` level now, and use a keyword `__xxxx` later on if we need to configure at the `Connection` level. Not saying it is required, just triggering a conversation about it.